src/opf/dcp.jl

```diff
@@ -1,263 +1,179 @@
+    set_upper_bound.(pg, gen_status .* pgmax)
+
+    # Flow bounds (both zero if branch is off)
+    set_lower_bound.(pf, branch_status .* -smax)
+    set_upper_bound.(pf, branch_status .* smax)
+
     # Slack bus
-    JuMP.@constraint(model, slack_bus, va[i0] == 0.0)
+    @constraint(model, slack_bus, va[i0] == 0.0)
 
     # Nodal power balance
-    JuMP.@constraint(model,
+    @expression(model, pt[e in 1:E], -pf[e])
+    @constraint(model,
         kirchhoff[i in 1:N],
-        sum(pg[g] for g in ref[:bus_gens][i])
-        - sum(pf[a] for a in ref[:bus_arcs][i]) 
-        ==
-        sum(load["pd"] for load in bus_loads[i])
-        + sum(shunt["gs"] for shunt in bus_shunts[i])
+        sum(gen_status[g] * pg[g] for g in bus_gens[i])
+        - sum(branch_status[a] * pf[a] for a in bus_arcs_fr[i])
+        - sum(branch_status[a] * pt[a] for a in bus_arcs_to[i])
+        == 
+        pd[i] + gs[i]
     )
 
-    # Branch power flow physics and limit constraints
-    # We pre-allocate constraint containers for simplicity
-    model[:voltage_difference_limit] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:ohm_eq] = Vector{JuMP.ConstraintRef}(undef, E)
-    for (i,branch) in ref[:branch]
-        data["branch"]["$i"]["br_status"] == 0 && continue  # skip branch
-
-        f_idx = (i, branch["f_bus"], branch["t_bus"])
-
-        p_fr = pf[f_idx]
+    model[:ohm] = Vector{ConstraintRef}(undef, E)
+    model[:va_diff] = Vector{ConstraintRef}(undef, E)
 
-        va_fr = va[branch["f_bus"]]
-        va_to = va[branch["t_bus"]]
-
-        g, b = PM.calc_branch_y(branch)
-
-        # From side of the branch flow
-        model[:ohm_eq][i] = JuMP.@constraint(model, b*(va_to - va_fr) - p_fr == 0)
+    for e in 1:E
+        # Branch power flow physics and limit constraints
+        model[:ohm][e] = @constraint(model,
+            branch_status[e] * (
+                -b[e] * (va[bus_fr[e]] - va[bus_to[e]])
+            ) - pf[e] == 0
+        )
 
         # Voltage angle difference limit
-        model[:voltage_difference_limit][i] = JuMP.@constraint(model, branch["angmin"] <= va_fr - va_to <= branch["angmax"])
+        model[:va_diff][e] = @constraint(model,
+            dvamin[e] ≤ branch_status[e] * (va[bus_fr[e]] - va[bus_to[e]]) ≤ dvamax[e]
+        )
     end
 
     #
     #   III. Objective
     #
-    l, u = extrema(gen["cost"][1] for (i, gen) in ref[:gen])
-    (l == u == 0.0) || @warn "Data $(data["name"]) has quadratic cost terms; those terms are being ignored"
-    JuMP.@objective(model, Min, sum(
-        gen["cost"][2]*pg[i] + gen["cost"][3]
-        for (i,gen) in ref[:gen]
-    ))
-
-    return OPFModel{DCOPF}(data, model)
-end
+    l, u = extrema(c2[g] for g in 1:G if gen_status[g])
+    (l == u == 0.0) || @warn "Data $(data.case) has quadratic cost terms; those terms are being ignored"
 
+    @objective(model,
+        Min,
+        sum(
+            gen_status[g] * (c1[g] * pg[g] + c0[g]) for g in 1:G
+        )
```

Comment:
For consistency with other formulations
```suggestion
        sum(c1[g] * pg[g] + c0[g] for g in 1:G if gen_status[g])
```